zeppelin-zengine/src/main/java/org/apache/zeppelin/notebook/NotebookAuthorization.java

```diff
@@ -31,17 +35,22 @@
  */
 public class NotebookAuthorization {
   private static final Logger LOG = LoggerFactory.getLogger(NotebookAuthorization.class);
-
+  private static NotebookAuthorization instance = null;
   /*
    * { "note1": { "owners": ["u1"], "readers": ["u1", "u2"], "writers": ["u1"] },  "note2": ... } }
    */
-  private Map<String, Map<String, Set<String>>> authInfo = new HashMap<>();
-  private ZeppelinConfiguration conf;
-  private Gson gson;
-  private String filePath;
+  private static Map<String, Map<String, Set<String>>> authInfo = new HashMap<>();
+  private static ZeppelinConfiguration conf;
+  private static Gson gson;
+  private static String filePath;
+
+  private NotebookAuthorization() {}
 
-  public NotebookAuthorization(ZeppelinConfiguration conf) {
-    this.conf = conf;
+  public static NotebookAuthorization init(ZeppelinConfiguration config) {
```

Comment:
@echarles good point. actually that's more of a singleton pattern that gets instantiated once on `init` and can be retrieved by `getInstance` further on. the purpose of it was to enable other modules (e.g. storage in zengine) not having direct reference to notebookAuthorization, to be able to use its services without changing api. I believe for longer term authorization and storage modules can be refactored in a way that we don't need that pattern.

regarding overriding the conf field, you're right. I addressed it in 9cf1d8848ee02b5fafda1afd538b71befd3eb412 so that it gets instantiated only once.